### PR TITLE
Add support for Microsoft.KubernetesConfiguration/extensions

### LINKED ISSUE #2934: Feature: Support for Azure container marketplace offerings (Microsoft.KubernetesConfiguration/extensions)
Add ability to install any extension such as [Azure container marketplace offers](https://learn.microsoft.com/en-us/azure/aks/deploy-marketplace) or Arc Extension (e.g. Flux v2) onto AKS clusters.

[Rest API is here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/stable/2022-11-01/extensions.json)

### PR DESCRIPTION
**Pull request by super-harsh**
Closes #2934 

**What this PR does / why we need it**:

This PR adds support for Microsoft.KubernetesConfiguration/extensions. 

_Also includes a generator fix to identify the extension types_

**Special notes for your reviewer**:
Removing the `ConfigurationProtectedSettings` configuration from the resource for now as it's dependent on #3435 being resolved. Once resolved, we can revert this field to be on the resource.

### CODE REVIEW

**Comment by super-harsh** [comment]
TODO: Need to add sample and record sample test

File: v2/api/kubernetesconfiguration/v1api20221101/structure.txt [hunk 1]
```diff
@@ -0,0 +1,267 @@
+// Code generated by azure-service-operator-codegen. DO NOT EDIT.
+github.com/Azure/azure-service-operator/v2/api/kubernetesconfiguration/v1api20221101
+├── APIVersion: Enum (1 value)
+│   └── "2022-11-01"
```
Comment (matthchr) [inline v2/api/kubernetesconfiguration/v1api20221101/structure.txt:4]: There's a 2023-05-01 API version, we should probably use that instead?

File: v2/api/kubernetesconfiguration/v1api20221101/structure.txt [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+│   │   │   ├── PromotionCode: *string
+│   │   │   ├── Publisher: *string
+│   │   │   └── Version: *string
+│   │   ├── ReleaseTrain: *string
+│   │   ├── Scope: *Object (2 properties)
+│   │   │   ├── Cluster: *Object (1 property)
+│   │   │   │   └── ReleaseNamespace: *string
+│   │   │   └── Namespace: *Object (1 property)
+│   │   │       └── TargetNamespace: *string
+│   │   ├── Statuses: Object (5 properties)[]
```
Comment (matthchr) [inline v2/api/kubernetesconfiguration/v1api20221101/structure.txt:31]: Hmm, can this field actually be set? Given its name I am wondering if it's really supposed to be marked readonly?

File: v2/azure-arm.yaml [hunk 3]
```diff
@@ -1587,6 +1599,11 @@ objectModelConfiguration:
       VirtualNetworkRule:
         Id:
           $armReference: true
+  kubernetesconfiguration:
+    2022-11-01:
```
Comment (matthchr) [inline v2/azure-arm.yaml:1603]: As mentioned above, 2023-05-01 is a newer API version than this -- should we use that instead?

File: v2/internal/controllers/samples_test.go [hunk 4]
```diff
@@ -39,6 +39,7 @@ var randomNameExclusions = []string{
 	"compute",
 	"documentdb",
 	"insights",
+	"kubernetesconfiguration",
```
Comment (matthchr) [inline v2/internal/controllers/samples_test.go:42]: minor: Why does this need to avoid random name exclusions?

Comment (super-harsh) [reply v2/internal/controllers/samples_test.go:42]: Cz the owner exists as a reference object, if we use the random names, there's no way the sample can find the owner if we use random names.

Comment (super-harsh) [reply v2/internal/controllers/samples_test.go:42]: Now if I think more of it.. its a tech-debt - I'll open an issue for this and add a todo on the exclusion.

File: v2/api/kubernetesconfiguration/v1api20221101/structure.txt (hunk 2 above)
Comment (matthchr) [reply v2/api/kubernetesconfiguration/v1api20221101/structure.txt]: Checking on this

File: v2/internal/controllers/samples_test.go (hunk 4 above)
Comment (matthchr) [reply v2/internal/controllers/samples_test.go:42]: Missing a TODO here?

File: v2/api/kubernetesconfiguration/v1api20221101/structure.txt (hunk 2 above)
Comment (super-harsh) [reply v2/api/kubernetesconfiguration/v1api20221101/structure.txt]: Done!

File: v2/internal/controllers/samples_test.go (hunk 4 above)
Comment (super-harsh) [reply v2/internal/controllers/samples_test.go:43]: Done!

**Review by matthchr — APPROVED** [review]
LGTM!

**Pull request merged**